### ISSUE TITLE
Спорадический баг в standalone варианте

### ISSUE THREAD

**Issue by Vol-2016** [Pmmlabs/OpenPeriscope#53]
Пока не нашел способа воспроизвести, но выглядит так:
нажатие одной из "кнопок" в чате - вдруг крешет OpenPeriscope.

(Если что - виндовский crash-dump+лог для одного из случаев сохранил)

**Comment by Vol-2016** [comment]
Не рецепт - нашел аналогию:
если убрать ffmpegsumo.dll и нажать (в чате или в другом месте) ссылку на видео - уже железно точно такой же креш. 
Может, что подсказывает?

**Comment by Pmmlabs** [comment]
не, второй случай совсем другой, он связан с тем что не обнаруживается то, что обязательно должно быть. А в первом случае крашится нодовский модуль ws, который отвечает за вебсокеты. у меня это тоже пару раз было, но я не стал разбираться, потому что баг был в коде модуля ws https://github.com/websockets/ws

**Comment by Vol-2016** [comment]
> версию ws

Привет. 
Подскажи, где в node_modules смотреть версию ws : 
node_modules\ws\package.json  - "version": "1.1.1"" в последней строчке, -  или где ?

**Comment by Pmmlabs** [comment]
"version": "1.1.0"
5 строчка

**Comment by Vol-2016** [comment]
Странно.
1) У меня ведь даже в исходном OpenPeriscope-1.5-windows-x86.zip (распакованном при запуске  в temp-папке) - не "1.1.0", а именно "1.1.1",
2) и в свежеполученном после 'npm install' (node-v6.9.1-x86) - тоже "1.1.1"...

И попутно:  структура дерева node_modules при 1) и 2) кейсах - здорово не совпадает.  Это в порядке вещей? Вроде и там и там все работает (разве что при 2) - креша пока не наблюдалось...)

В общем, посмотри, пожалуйста, аттач, там 1) + 2).

**Comment by Pmmlabs** [comment]
а, ну значит npm берет не из master ветки гитхаба. Я просто зашел на страничку репозитория и посмотрел.
структура должна быть одинаковая, я же собираю как - делаю npm install, и пакую все файлы, в том числе node_modules, в zip архив, и склеиваю с бинарником nw, т.е. я никуда ничего не перемещаю перед паковкой.
Структура различается, да, во втором случае почему-то зависимости ws лежат рядом с самим ws, наверное от версии node зависит, у меня 4 помоему, но не последняя точно. И на работу это не должно влиять

**Comment by Vol-2016** [comment]
Все-таки версия ws может как быть, так и не быть причиной креша, выше я уже привлекал внимание к ffmpegsumo.dll, с ней есть непонятки. 

Вот как у меня на Win7 выглядит папка OpenPeriscope и папка с nwjs-0.12.3 (на Win7):

![op_nwjs-0123](https://cloud.githubusercontent.com/assets/18728512/19827676/e8b0c362-9dba-11e6-9a68-39cc3a4bbc38.png)

**Comment by Pmmlabs** [comment]
Если ты намекаешь, что причина в каких-то отсутствующих файлах, то ты ошибаешься. В случае с ffmpegsumo.dll из-за отсутствующего файла функционал вообще не работал, тут же он работает, но падает из за некоторой конкретной ошибки

**Comment by Vol-2016** [comment]
я еще не дописал, сейчас отвлекли, но вскоре докончу :)

**Comment by Vol-2016** [comment]
Так, продолжаю.
Почему у меня под подозрением ffmpegsumo.dll ? 
А разве ты не потому заменил ffmpegsumo.dll из дистрибутива nwjs-v0.12.3-win-ia32.zip, что c OpenPeriscope она работала паршиво (видео практически не принималось), хотя и лежала рядом, не отсутствовала? Так есть ли гарантия, что найденная и вроде бы работающая версия не содержит и другие, более скрытые баги? (Я, кстати, и сам перепробовал кучу версий ffmpegsumo.dll: с OpenPeriscope половина не пашет).

Что касается отсутствующих файлов: специально в папку OpenPeriscope положил nw.exe (а не линк \nwjs\nw.exe), чтоб тупо убедиться, что отсутствие nwjc.exe, pdf.dll, libGLESv2.dll, libEGL.dll, d3dcompiler_47.dll - не влияет...
***
Кстати, в инструкции "Using as standalone application" строчка 
> 3. Download and unpack ffmpeg library to nw.js directory
- в случае nwjs-0.12.3 вроде лишняя?  ffmpeg.dll же не используется (как и вообще вся папка \nwjs\ целиком) ?

**Comment by Pmmlabs** [comment]
Да, это надо поправить, ссылка осталась от предыдущей инструкции для nw > 12, теперь копировать надо ffmpegsumo.dll из дистрибутива Google chrome. Эта библиотека нужна только для декодирования кодека h264, больше ни для чего. А уж на чат тем более не влияет. Описанный баг связан только с модулем ws

**Comment by Pmmlabs** [comment]
@Vol-2016 удали папку node_modules, замени в package.json `"ws": "*"` на `"ws": "1.1.0"`, сделай npm install и проверь, будет ли воспроизводиться проблема

**Comment by Vol-2016** [comment]
Сделал для unpacked варианта EXE (остается ждать - спорадический же :) )

Но для полноты и достоверности тестирования - 
все-таки пришли еще и свой модифицированный вариант  EXE, взамен релизному версии 1.6.
